tee-worker/litentry/core/vc-issuance/lc-vc-task-receiver/src/lib.rs

```diff
@@ -39,24 +38,56 @@ use itp_sgx_crypto::{ShieldingCryptoDecrypt, ShieldingCryptoEncrypt};
 use itp_sgx_externalities::SgxExternalitiesTrait;
 use itp_stf_executor::traits::StfEnclaveSigning;
 use itp_stf_state_handler::handle_state::HandleState;
-use itp_storage::{storage_map_key, StorageHasher};
+use itp_storage::{storage_map_key, storage_value_key, StorageHasher};
 use itp_top_pool_author::traits::AuthorApi;
 use itp_types::parentchain::ParentchainId;
 use lc_stf_task_receiver::StfTaskContext;
 use lc_stf_task_sender::AssertionBuildRequest;
-use lc_vc_task_sender::init_vc_task_sender_storage;
+use lc_vc_task_sender::{init_vc_task_sender_storage, TrustedVCRequestSigned};
 use litentry_primitives::{
 	aes_decrypt, AesOutput, Identity, IdentityNetworkTuple, RequestAesKey, ShardIdentifier,
 };
 use pallet_identity_management_tee::{identity_context::sort_id_graph, IdentityContext};
 use std::{
+	collections::HashMap,
 	format,
 	string::{String, ToString},
 	sync::Arc,
+	time::{Duration, Instant},
 	vec::Vec,
 };
 use threadpool::ThreadPool;
 
+#[cfg(feature = "std")]
+use std::sync::RwLock;
+
+#[cfg(feature = "sgx")]
+use std::sync::SgxRwLock as RwLock;
+
+pub struct RateLimiter {
+	requests: RwLock<HashMap<String, Instant>>,
+}
+
+impl RateLimiter {
+	fn new() -> Self {
```

Comment:
Yeah Ideally I want a bit more sophisticated cooldown period, For example, If someone's request returns a specific  error then the cooldown period is depending on that type of error, I think it would take some time to implement such a mechanism. So yeah we need a consensus on the arbitrary number, How does a 2 second period sound like for all cases for now?

Reply:
I think this will also depend on the implementation for this issue - https://linear.app/litentry/issue/P-298/implement-an-abortstrategy-for-assertion-building

Reply:
Yes that takes some time to implement. 2s is fine as the initial configuration